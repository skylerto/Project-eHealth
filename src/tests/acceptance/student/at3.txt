 -- at3.txt\
add_physician	   (1, "Mayo", specialist)
add_physician	   (2, "Mayo", specialist)
add_physician	   (3, "Mayo", generalist)

add_patient        (3, "Drew")
add_patient        (1, "Drew")

add_medication     (1, ["Wafarin", pill, 1.0, 6.0])
add_medication     (2, ["acetaminophen", liquid, 1.0, 25.5])

new_prescription   (2, 1, 3)
new_prescription   (4, 3, 1)
new_prescription   (1, 1, 1)

new_prescription   (3, 2, 3)

add_medicine       (1, 1, 5.5)
add_medicine       (1, 2, 5.5)

add_medicine       (4, 2, 5.5)

add_medicine       (2, 2, 5.5)
add_medicine       (2, 1, 5.5)

add_medicine       (3, 1, 5.5)
add_medicine       (3, 2, 5.5)

-- error cases
-- add_physician
add_physician	   (-1, "Mayo", specialist)
add_physician	   (1, "Mayo", specialist)
add_physician	   (6, "99Yo", specialist)

-- add_patient
add_patient        (-1, "Drew")
add_patient        (1, "Drew")
add_patient        (7, "76rew")

-- add_medication
add_medication     (-1, ["Wafarin", pill, 1.0, 6.0])
add_medication     (1, ["Wafarin", pill, 1.0, 6.0])
add_medication     (6, ["23afarin", pill, 1.0, 6.0])
add_medication     (7, ["Wafarin", pill, 1.0, 6.0])
add_medication     (8, ["Wafarin", pill, 6.0, 1.0])

dpr_q
-- add_interaction
add_interaction    (1,-2)
add_interaction    (1,1)
add_interaction    (7,9)
add_interaction    (1,2)
add_interaction    (1,2)
add_interaction    (2,1)
dpr_q

-- new_prescription
new_prescription   (-3, 2, 3)
new_prescription   (4, 2, 3)
new_prescription   (5, -2, 3)
new_prescription   (6, 2, 3)
new_prescription   (7, 2,-3)
new_prescription   (8, 2, 3)
new_prescription   (3, 20, 3)
new_prescription   (3, 2, 30)

-- add_medicine
add_medicine       (-2, 1, 5.5)
add_medicine       (2, -1, 5.5)
add_medicine       (2, 100, 5.5)
add_medicine       (2, 1, 5.5)
add_medicine       (2, 1, 5.5)
add_medicine       (2, 100, 5500)

prescriptions_q(-1000)
prescriptions_q(1000)
