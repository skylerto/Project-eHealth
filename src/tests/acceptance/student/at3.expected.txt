  0: ok
  Physicians:
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(1,"Mayo",specialist)
  1: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(3,"Drew")
  2: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(1,"Helen")
  3: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_medication(1,["Wafarin", pill, 1, 6])
  4: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
  Interactions: 
  Prescriptions:
->add_medication(3,["caffeine", liquid, 1, 16])
  5: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
  Prescriptions:
->add_medication(2,["acetaminophen", liquid, 1, 25.5])
  6: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
  Prescriptions:
->add_interaction(1,2)
  7: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
  Prescriptions:
->add_interaction(1,3)
  8: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
  Prescriptions:
->add_interaction(2,3)
  9: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
->new_prescription(2,1,3)
  10: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    2->[2,1,3,()]
->new_prescription(1,1,1)
  11: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
->add_medicine(1,1,5.5)
  12: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
->add_medicine(1,2,5.5)
  13: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,()]
->add_medicine(1,3,5.5)
  14: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,()]
->add_medicine(2,2,5.5)
  15: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,(2->5.50)]
->add_medicine(2,3,5.5)
  16: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,(2->5.50,3->5.50)]
->add_medicine(2,1,5.5)
  17: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,(1->5.50,2->5.50,3->5.50)]
->dpr_q
  18: ok
  There are dangerous prescriptions:
    (Helen,1)->{ [Wafarin,pl,1]->[acetaminophen,lq,2],[Wafarin,pl,1]->[caffeine,lq,3],[acetaminophen,lq,2]->[caffeine,lq,3] }
    (Drew,3)->{ [Wafarin,pl,1]->[acetaminophen,lq,2],[Wafarin,pl,1]->[caffeine,lq,3],[acetaminophen,lq,2]->[caffeine,lq,3] }
->remove_medicine(2,1)
  19: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,(2->5.50,3->5.50)]
->remove_medicine(2,2)
  20: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,(3->5.50)]
->remove_medicine(2,3)
  21: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,()]
->add_medicine(1,1,5.5)
  22: medication is already prescribed
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,()]
->add_medicine(1,2,5.5)
  23: medication is already prescribed
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,()]
->add_medicine(1,3,5.5)
  24: medication is already prescribed
  Physicians:
    1->[Mayo,sp]
  Patients: 
    1->Helen
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
    3->[caffeine,lq,1.00,16.00]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
    [Wafarin,pl,1]->[caffeine,lq,3]
    [acetaminophen,lq,2]->[caffeine,lq,3]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50,3->5.50)]
    2->[2,1,3,()]
->dpr_q
  25: ok
  There are dangerous prescriptions:
    (Helen,1)->{ [Wafarin,pl,1]->[acetaminophen,lq,2],[Wafarin,pl,1]->[caffeine,lq,3],[acetaminophen,lq,2]->[caffeine,lq,3] }
